src/Npgsql/NpgsqlDataSource.cs

```diff
@@ -247,6 +248,68 @@ public async Task ReloadTypesAsync(CancellationToken cancellationToken = default
         }
     }
 
+    /// <summary>
+    /// Attempts to return a mapping for a specific type.
+    /// </summary>
+    public PgTypeInfo? TryGetMapping<T>(string? dataTypeName = null) => TryGetMapping(typeof(T), dataTypeName);
+
+    /// <summary>
+    /// Attempts to return a mapping for a specific type.
+    /// </summary>
+    public PgTypeInfo? TryGetMapping(Type? type = null, string? dataTypeName = null)
+    {
+        if (type is null && string.IsNullOrEmpty(dataTypeName))
+            throw new ArgumentException($"Either {nameof(type)} or {nameof(dataTypeName)} must be specified.");
+
+        // ReSharper disable once ConditionIsAlwaysTrueOrFalseAccordingToNullableAPIContract
+        if (SerializerOptions is null)
+            using (OpenConnection()) { }
+
+        return TryGetMappingCore(type, dataTypeName);
+    }
+
+    /// <summary>
+    /// Attempts to return a mapping for a specific type.
+    /// </summary>
+    public ValueTask<PgTypeInfo?> TryGetMappingAsync<T>(string? dataTypeName = null, CancellationToken cancellationToken = default)
+        => TryGetMappingAsync(typeof(T), dataTypeName, cancellationToken);
+
+    /// <summary>
+    /// Attempts to return a mapping for a specific type.
+    /// </summary>
+    public async ValueTask<PgTypeInfo?> TryGetMappingAsync(Type? type = null, string? dataTypeName = null, CancellationToken cancellationToken = default)
```

Comment:
Yeah, ideally we'd be able to return mapping info to the user without ever connecting to the database, purely based on how the data source was configured - in which case we wouldn't need an async API here. This is a bit related to us returning a separate type (not PgTypeInfo), which would presumably return restricted info that's purely client-side mapping config.

Or am I missing something and this mapping information API really doesn't make sense without first connecting to the database and getting actual type OIDs etc.?